examples/tutorials/lighting_tutorial.py

```diff
@@ -218,6 +221,78 @@ def main(show_imgs=True, save_imgs=False):
 
     # [/example 9]
 
+    remove_all_objects(sim)
+
+    # [example 10]
+
+    # set a new default light setup
+    my_default_lighting = [
+        LightInfo(position=[8.0, 8.0, 8.0], model=LightPositionModel.GLOBAL),
+        LightInfo(position=[8.0, 8.0, -8.0], model=LightPositionModel.GLOBAL),
+        LightInfo(position=[-8.0, 8.0, 8.0], model=LightPositionModel.GLOBAL),
+        LightInfo(position=[-8.0, 8.0, -8.0], model=LightPositionModel.GLOBAL),
+    ]
+    sim.set_light_setup(my_default_lighting)
+
+    # add a box
+    id_1 = sim.add_object(box_template_id)
+    sim.set_translation([3.2, 0.23, 0.4], id_1)
+
+    get_obs(sim, show_imgs, save_imgs)
+
+    # The box render asset uses different materials for its faces. There are
+    # three materials in total, named "Material", "Material.001", and
+    # "Material.002" in the GLB file. Let's change Material.002.
+
+    # get the render asset for the box template
+    render_asset_handle = sim.get_object_initialization_template(
+        id_1
+    ).render_asset_handle
+
+    # create a new material: purple with a yellow specular highlight.
+    new_material = PhongMaterialInfo(
+        ambient_color=mn.Color4(0.15, 0.0, 0.15, 1),
+        diffuse_color=mn.Color4(0.15, 0.0, 0.15, 1),
+        specular_color=mn.Color4(1.0, 1.0, 0.0, 1),
```

Comment:
These could be all just lists / tuples I think, the actual type is not important:

```suggestion
        ambient_color=(0.15, 0.0, 0.15, 1),
        diffuse_color=(0.15, 0.0, 0.15, 1),
        specular_color=(1.0, 1.0, 0.0, 1),
```

... and maybe even with the alpha omitted, but I'm not 100% sure if pybind handles such a deep conversion correctly. Same elsewhere if there's more similar cases.

Reply:
^Pretty sure it can actually, just might need some massaging to get it to work. You can definitely construct mn.Color4 for all the values in the tuples within the pybind lambda function. At the very least, PyBind can convert tuple to std::tuple and then you can to do the conversion in C++. You could also create a constructor fro mn.Color4 takes in a std::tuple etc.. Finally, you can just implement the cast yourself if needed: https://pybind11.readthedocs.io/en/stable/advanced/cast/custom.html

Reply:
Yeah I mean all things for this conversion (tuple constructors, implicit conversion marks etc.) should be in place, so I'm 90% sure it'll work -- but not 100% because pybind is impenetrable magic :))